src/FlashLoanTracker.sol

```diff
@@ -0,0 +1,55 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8;
+
+import {ICowSettlement} from "./interface/ICowSettlement.sol";
+import {IERC20} from "./vendored/IERC20.sol";
+import {SafeERC20} from "./vendored/SafeERC20.sol";
+
+library FlashLoanTrackerError {
+    error AlreadyTakenOut();
+    error NotTakenOut();
+}
+
+/// @title Flash Loan Tracker
+/// @author CoW DAO developers
+/// @notice A contract that tracks flash loan takeouts and paybacks for the AaveBorrower.
+contract FlashLoanTracker {
+    using SafeERC20 for IERC20;
+
+    ICowSettlement public immutable settlementContract;
+    address public immutable borrower;
```

Comment:
I don't think having one standalone tracker per borrower is a good design. We should either make the tracker borrower agnostic (and have it as part of the arguments in `takeOut`) or make it part of the `Borrower` base implementation directly. This way we avoid configuration nightmare where we need to store both a borrower and a tracker instance pre flashloan provider.